packages/core/src/types/mitosis-node.ts

```diff
@@ -27,6 +27,10 @@ export type MitosisNode = {
    * }
    * ```
    */
-  bindings: { [key: string]: string | undefined };
+  bindings: {
+    [key: string]:
+      | { code: string | undefined; arguments?: string[] }
+      | undefined;
+  };
```

Comment:
This type information isn't quite accurate. the `code` is always going to be defined. the `undefined` possibility bubbles up to the overall object now:

```suggestion
  bindings: {
    [key: string]:
      | { code: string; arguments?: string[] }
      | undefined;
  };
```

If we make this change, then a lot of type casts that were added in this PR become unnecessary.

@decadef20 If possible, it would be great to make a follow-up PR that makes this improvement!

Reply:
Thanks for your CR. I'll fix it later

Reply:
<img width="1140" alt="image" src="https://user-images.githubusercontent.com/7866718/169684182-c5333840-51ff-4a23-b5c8-7fd3a27ab4a0.png">

I change the `binding` type, and delete type cast,  but compile fails. 

`binding[key]` maybe `undefined`,  It seems like we have to use type cast here.

Reply:
If it's only thats the only place where there's a problem then you can change it